src/NServiceBus.Core/Outbox/OutboxRecordBehavior.cs

```diff
@@ -6,7 +6,7 @@ namespace NServiceBus
 
     class OutboxRecordBehavior : PhysicalMessageProcessingStageBehavior
     {
-        public IOutboxStorage OutboxStorage { get; set; }
+        public IStoreOutboxMessages OutboxStorage { get; set; }
```

Comment:
To align with the other behavior and also follow what we are heading towards I'd say we should use ctor injection here as well.